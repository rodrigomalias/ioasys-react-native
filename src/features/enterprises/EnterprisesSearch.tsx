import * as React from "react"
import { StyleSheet, Text, TextInput, TouchableOpacity, View } from "react-native"
import { firstLetter, translation } from "../../utils"
import { colors } from "../../colors"
import { Picker } from "@react-native-picker/picker"
import { enterpriseTypes, IEnterpriseType } from "./Enterprises.utils"
import { Icon } from "react-native-elements"

interface IEnterpriseSearch {
    callback: Function
}

export const EnterprisesSearch = (props: IEnterpriseSearch) => {
    const { callback } = props
    const [ name, setName ] = React.useState("")
    const [ type, setType ] = React.useState(0)

    const handleSearch = React.useCallback(() => {

        const typeFilter = type === 0 ? null : type

        callback(typeFilter, name)
    }, [ callback, name, type ])

    return (
        <View style={styles.containerHeader}>
            <View style={styles.inputView}>
                <Picker
                    testID={"test_enterprises_search_picker"}
                    selectedValue={type}
                    style={styles.input}
                    onValueChange={(itemValue: React.SetStateAction<number>) => setType(itemValue)}
                >
                    <Picker.Item label={firstLetter(translation("type"))} value={0} />
                    {enterpriseTypes.map((enterpriseType: IEnterpriseType) => {
                        return (
                            <Picker.Item
                                label={enterpriseType.typeName}
                                value={enterpriseType.type}
                                key={enterpriseType.type} />
                        )
                    })}
                </Picker>
            </View>
            <View style={styles.inputView}>
                <TextInput
                    testID={"test_enterprises_search_input"}
                    value={name}
                    style={styles.input}
                    autoCompleteType={"off"}
                    placeholder={firstLetter(translation("name"))}
                    onChangeText={(name) => setName(name)}/>
            </View>
            <TouchableOpacity
                testID={"test_enterprises_search_button"}
                style={styles.searchButton}
                onPress={handleSearch}>
                <Text style={styles.searchButtonText}>
                    <Icon
                        name='search'
                        type='font-awesome'
                        size={20}
                        color={colors.whiteBackground}/>
                </Text>
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    containerHeader: {
        padding: 5,
        paddingTop: 20,
        paddingBottom: 20,
        borderBottomColor: colors.darkGray,
        borderBottomWidth: 0.5,
        flexDirection: "row",
        justifyContent: "center",
        backgroundColor: colors.whiteBackground,
        shadowColor: colors.black,
        shadowOffset: {
            width: 2,
            height: 2,
        },
        shadowOpacity: 1,
        shadowRadius: 3.84,
        elevation: 5,
    },
    searchButton: {
        width: 50,
        height: 50,
        borderRadius: 25,
        backgroundColor: colors.primaryDark,
        marginLeft: 10,
        marginRight: 10,
        textAlign: "center",
        justifyContent: "center",
    },
    searchButtonText: {
        alignSelf: "center",
        color: colors.white,
    },
    inputView: {
        flex: 1,
        height: 45,
        justifyContent: "center",
        borderWidth: 0.5,
        borderRadius: 5,
        backgroundColor: colors.white,
        margin: 5,
    },
    input: {
        width: "100%",
    },
})