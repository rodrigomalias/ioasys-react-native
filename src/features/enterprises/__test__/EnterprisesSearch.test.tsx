import React from "react"
import { EnterprisesSearch } from "../EnterprisesSearch"
import { fireEvent, render } from "@testing-library/react-native"
import renderer from "react-test-renderer"

const mockValue = {
    city: "SP",
    country: "Brazil",
    description: "Description test",
    email_enterprise: "email@email.com",
    enterprise_name: "Company test",
    enterprise_type: {
        enterprise_type_name: "Type test",
        id: 1
    },
    facebook: undefined,
    id: 1,
    linkedin: undefined,
    own_enterprise: false,
    phone: undefined,
    photo: "photo.png",
    share_price: 5000,
    shares: undefined,
    twitter: undefined,
    value: 500
}

const initialState = {
    getEnterpriseById: () => jest.fn(),
    getEnterprisesNameOrType: () => jest.fn(),
    getEnterprises: () => jest.fn(),
    enterprise: mockValue,
    enterprises: [ mockValue ],
    errorEnterprise: "",
    errorEnterprises: "",
}

describe("<EnterprisesSearch/>", () => {
    test("should render correctly", () => {
        const tree = renderer.create(<EnterprisesSearch callback={() => jest.fn()}/>).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test("should find the picker using testId", () => {
        const { getByTestId } = render(<EnterprisesSearch callback={() => jest.fn()}/>)

        const picker = getByTestId("test_enterprises_search_picker")

        expect(picker).not.toBeNull()
    })

    test("should find the input using testId", () => {
        const { getByTestId } = render(<EnterprisesSearch callback={() => jest.fn()}/>)

        const input = getByTestId("test_enterprises_search_input")

        expect(input).not.toBeNull()
    })

    test("input has text", () => {
        const { getByTestId } = render(<EnterprisesSearch callback={() => jest.fn()}/>)
        const mockName = "enterprise name"

        const input = getByTestId("test_enterprises_search_input")
        fireEvent.changeText(input, mockName)

        expect(input.props.value).toBe(mockName)
    })

    test("should find the button using testId", () => {
        const { getByTestId } = render(<EnterprisesSearch callback={() => jest.fn()}/>)

        const button = getByTestId("test_enterprises_search_button")

        expect(button).not.toBeNull()
    })
})