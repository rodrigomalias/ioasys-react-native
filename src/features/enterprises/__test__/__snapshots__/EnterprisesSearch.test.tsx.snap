// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnterprisesSearch/> should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "borderBottomColor": "#999",
      "borderBottomWidth": 0.5,
      "elevation": 5,
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 5,
      "paddingBottom": 20,
      "paddingTop": 20,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 2,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flex": 1,
        "height": 45,
        "justifyContent": "center",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <RNCPicker
        items={
          Array [
            Object {
              "label": "Tipo",
              "testID": undefined,
              "textColor": undefined,
              "value": 0,
            },
            Object {
              "label": "Agro",
              "testID": undefined,
              "textColor": undefined,
              "value": 1,
            },
            Object {
              "label": "Aviation",
              "testID": undefined,
              "textColor": undefined,
              "value": 2,
            },
            Object {
              "label": "Biotech",
              "testID": undefined,
              "textColor": undefined,
              "value": 3,
            },
            Object {
              "label": "Eco",
              "testID": undefined,
              "textColor": undefined,
              "value": 4,
            },
            Object {
              "label": "Ecommerce",
              "testID": undefined,
              "textColor": undefined,
              "value": 5,
            },
            Object {
              "label": "Education",
              "testID": undefined,
              "textColor": undefined,
              "value": 6,
            },
            Object {
              "label": "Fashion",
              "testID": undefined,
              "textColor": undefined,
              "value": 7,
            },
            Object {
              "label": "Fintech",
              "testID": undefined,
              "textColor": undefined,
              "value": 8,
            },
            Object {
              "label": "Food",
              "testID": undefined,
              "textColor": undefined,
              "value": 9,
            },
            Object {
              "label": "Games",
              "testID": undefined,
              "textColor": undefined,
              "value": 10,
            },
            Object {
              "label": "Health",
              "testID": undefined,
              "textColor": undefined,
              "value": 11,
            },
            Object {
              "label": "IOT",
              "testID": undefined,
              "textColor": undefined,
              "value": 12,
            },
            Object {
              "label": "Logistics",
              "testID": undefined,
              "textColor": undefined,
              "value": 13,
            },
            Object {
              "label": "Media",
              "testID": undefined,
              "textColor": undefined,
              "value": 14,
            },
            Object {
              "label": "Mining",
              "testID": undefined,
              "textColor": undefined,
              "value": 15,
            },
            Object {
              "label": "Products",
              "testID": undefined,
              "textColor": undefined,
              "value": 16,
            },
            Object {
              "label": "Real Estate",
              "testID": undefined,
              "textColor": undefined,
              "value": 17,
            },
            Object {
              "label": "Service",
              "testID": undefined,
              "textColor": undefined,
              "value": 18,
            },
            Object {
              "label": "Smart City",
              "testID": undefined,
              "textColor": undefined,
              "value": 19,
            },
            Object {
              "label": "Social",
              "testID": undefined,
              "textColor": undefined,
              "value": 20,
            },
            Object {
              "label": "Software",
              "testID": undefined,
              "textColor": undefined,
              "value": 21,
            },
            Object {
              "label": "Technology",
              "testID": undefined,
              "textColor": undefined,
              "value": 22,
            },
            Object {
              "label": "Tourism",
              "testID": undefined,
              "textColor": undefined,
              "value": 23,
            },
            Object {
              "label": "Transport",
              "testID": undefined,
              "textColor": undefined,
              "value": 24,
            },
          ]
        }
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
        testID="test_enterprises_search_picker"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flex": 1,
        "height": 45,
        "justifyContent": "center",
        "margin": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="off"
      onChangeText={[Function]}
      placeholder="Nome"
      rejectResponderTermination={true}
      style={
        Object {
          "width": "100%",
        }
      }
      testID="test_enterprises_search_input"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#01814e",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginLeft": 10,
        "marginRight": 10,
        "opacity": 1,
        "textAlign": "center",
        "width": 50,
      }
    }
    testID="test_enterprises_search_button"
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFFFFF",
        }
      }
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#F5FCFF",
                    "fontSize": 20,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              ï€‚
            </Text>
          </View>
        </View>
      </View>
    </Text>
  </View>
</View>
`;
